packages/cozy-stack-client/src/OAuthClient.js

```diff
@@ -287,14 +287,19 @@ class OAuthClient extends CozyStackClient {
   getAuthCodeURL(stateCode, scopes = this.scope) {
     if (!this.isRegistered()) throw new NotRegisteredException()
 
-    const query = {
+    let query = {
       client_id: this.oauthOptions.clientID,
       redirect_uri: this.oauthOptions.redirectURI,
       state: stateCode,
       response_type: 'code',
       scope: scopes.join(' ')
     }
-
+    if (this.oauthOptions.registerToken) {
```

Comment:
nit: pas de garde-fou style `this.oauthOptions?.registerToken` ? `this.oauthOptions` est forcément defined ?

Reply:
oui